include/gsl/util

```diff
@@ -110,6 +111,7 @@ GSL_SUPPRESS(bounds.2) // NO-FORMAT: attribute
     // clang-format on
     constexpr T& at(T (&arr)[N], const index i)
 {
+    static_assert(N <= static_cast<std::size_t>(std::numeric_limits<std::ptrdiff_t>::max()), "We only support arrays up to half the bytes of the address space.");
```

Comment:
with the latest changes, the `static_assert` message is less clear - perhaps instead of "half the bytes..." it should just be "`PTRDIFF_MAX` elements" (or something similar)